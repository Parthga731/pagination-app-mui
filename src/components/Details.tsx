import { useEffect } from "react";
import { useLocation } from "react-router-dom";

export const Details = () => {
  const location = useLocation();
  useEffect(() => {
    console.log(location.state);
  }, [location.state]);

  return (
    <>
      <h1 data-testid="Details-heading"> user Information</h1>
      {JSON.stringify(location.state)}
    </>
  );
};
