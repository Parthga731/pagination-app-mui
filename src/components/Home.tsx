import React, { useEffect, useState } from "react";
import { useAppDispatch, useAppSelector } from "../app/hooks";
import { getPagesData, showPagination } from "../redux/PaginationSlice";
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Paper,
  TextField,
  Pagination,
  Stack,
  Box,
} from "@mui/material";
import moment from "moment";
import { useNavigate } from "react-router-dom";

export const Home = () => {
  const { data } = useAppSelector((state) => state.pages);
  const dispatch = useAppDispatch();
  const navigate = useNavigate();
  console.log(data);

  const [page, setPage] = useState(0);
  const [pageValue, setpageValue] = useState(0);
  const [searchValue, setSearchValue] = useState<string>("");

  //   useEffect(() => {
  //     axios
  //       .get(
  //         `https://hn.algolia.com/api/v1/search_by_date?tags=story&page=${page}`
  //       )
  //       .then((result) => {
  //         console.log(result);
  //         dispatch(addData(result.data));
  //         // dispatch(showPagination());
  //       })
  //       .catch((error) => console.log(error));
  //   }, [page, dispatch]);

  useEffect(() => {
    dispatch(getPagesData(page));
    dispatch(showPagination());
  }, [dispatch, page]);

  useEffect(() => {
    const interval = setTimeout(() => {
      setPage(page + 1);
    }, 10000);
    return () => clearInterval(interval);
  }, [page]);

  const changePage = (event: any, value: any) => {
    setpageValue(value - 1);
  };

  const handleClick = (row: any) => {
    navigate("./details", { state: { row } });
  };

  return (
    <>
      <Box data-test="component-pagination">
        <Paper
          elevation={5}
          sx={{
            p: 5,
            display: "flex",
            flexDirection: "column",
            justifyContent: "center",
            alignItems: "center",
          }}>
          <TextField
            id="filled-basic"
            label="Search by Title and created at"
            variant="standard"
            onChange={(e: any) => setSearchValue(e.target.value)}
          />

          <Table sx={{ width: 850 }} aria-label="simple table">
            <TableHead>
              <TableRow>
                <TableCell>ID</TableCell>
                <TableCell>Created_at</TableCell>
                <TableCell>Title</TableCell>
                <TableCell>Author</TableCell>
                <TableCell>url</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {/* {console.log(data[pageValue]?.hits)} */}
              {data[pageValue]?.hits
                ?.filter(
                  (info: any) =>
                    info.title
                      .toUpperCase()
                      .includes(searchValue.toUpperCase()) ||
                    info.created_at
                      .toUpperCase()
                      .includes(searchValue.toUpperCase())
                )
                ?.map((row: any, index: number) => {
                  return (
                    <TableRow key={index} onClick={() => handleClick(row)}>
                      <TableCell>{index + 1}</TableCell>
                      <TableCell>
                        {moment(row.created_at).format("LLLL")}
                      </TableCell>
                      <TableCell>{row.title}</TableCell>
                      <TableCell>{row.author}</TableCell>
                      <TableCell>{row.url}</TableCell>
                    </TableRow>
                  );
                })}
            </TableBody>
          </Table>
          <Stack spacing={2}>
            <Pagination
              count={data.length}
              page={pageValue + 1}
              onChange={changePage}
              color="primary"
              sx={{ m: 3 }}
              showFirstButton
              showLastButton
            />
          </Stack>
        </Paper>
      </Box>
    </>
  );
};
