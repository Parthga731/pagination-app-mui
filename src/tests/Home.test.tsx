import React from "react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../app/store";
import ReactDOM from "react-dom";
import { Home } from "../components/Home";

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

describe("testing home snapshot", () => {
  let container: HTMLDivElement;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
    ReactDOM.render(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>,
      container
    );
  });
  afterEach(() => {
    document.body.removeChild(container);
    container.remove();
  });

  it("render correctly", () => {
    expect(container).toMatchSnapshot();
  });
});

// test("render without crashing", () => {
//   const wrapper = mount(
//     <BrowserRouter>
//       <Provider store={store}>
//         <Home />
//       </Provider>
//     </BrowserRouter>
//   );
//   const appComponent = wrapper.find("[data-test='component-pagination']");
//   // console.log(wrapper.debug())
//   expect(appComponent.length).toBe(1);
// });
