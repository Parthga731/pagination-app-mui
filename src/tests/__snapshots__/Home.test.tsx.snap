// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing home snapshot render correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
    data-test="component-pagination"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-13nwz9e-MuiPaper-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="filled-basic"
          id="filled-basic-label"
        >
          Search by Title and created at
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id="filled-basic"
            type="text"
            value=""
          />
        </div>
      </div>
      <table
        aria-label="simple table"
        class="MuiTable-root css-hnal74-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              ID
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Created_at
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Title
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Author
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              url
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        />
      </table>
      <div
        class="css-1p5q5e5-MuiStack-root"
      >
        <nav
          aria-label="pagination navigation"
          class="MuiPagination-root MuiPagination-text css-3k3vc1-MuiPagination-root"
        >
          <ul
            class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
          >
            <li>
              <button
                aria-label="Go to first page"
                class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-firstLast css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                  data-testid="FirstPageIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                  />
                </svg>
              </button>
            </li>
            <li>
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                  data-testid="NavigateBeforeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  />
                </svg>
              </button>
            </li>
            <li>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                  data-testid="NavigateNextIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </button>
            </li>
            <li>
              <button
                aria-label="Go to last page"
                class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-firstLast css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                  data-testid="LastPageIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                  />
                </svg>
              </button>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
